packages/arb-validator-core/gobridge/challenge.go

```diff
@@ -30,39 +31,45 @@ const (
 	challengerTurn = 2
 )
 
+type challengeData struct {
+	deadline             common.TimeTicks
+	challengerDataHash   common.Hash
+	state                int
+	challengePeriodTicks common.TimeTicks
+	asserter             common.Address
+	challenger           common.Address
+	challengeType        *big.Int
+}
+
 type challenge struct {
 	client          *GoArbAuthClient
 	challengeData   *challengeData
 	contractAddress common.Address
 }
 
 func newChallenge(address common.Address, client *GoArbAuthClient) (*challenge, error) {
-	chal := challenge{
-		client:          client,
-		contractAddress: address,
-		challengeData:   client.challenges[address],
-	}
-
-	return &chal, nil
+	return client.challenges[address], nil
```

Comment:
This function currently never returns an error. Either the signature should be changed or error conditions should be added. It'd be good to have a consistent behavior for how to handle contracts that don't exist. I believe that go would return a `nil` here if the address didn't exist in that map.

Reply:
cleaned up error handling